src/Runner.Worker/Handlers/CompositeActionHandler.cs

```diff
@@ -113,69 +108,57 @@ private void ProcessCompositeActionOutputs()
                 DictionaryContextData actionOutputs = actionManifestManager.EvaluateCompositeOutputs(ExecutionContext, Data.Outputs, evaluateContext);
 
-                // Set the outputs for the outputs object in the whole composite action
-                // Each pair is structured like this
-                // We ignore "description" for now
-                // {
-                //   "the-output-name": {
-                //     "description": "",
-                //     "value": "the value"
-                //   },
-                //   ...
-                // }
+                // Set outputs
+                //
+                // Each pair is structured like:
+                //   {
+                //     "the-output-name": {
+                //       "description": "",
+                //       "value": "the value"
+                //     },
+                //     ...
+                //   }
                 foreach (var pair in actionOutputs)
                 {
-                    var outputsName = pair.Key;
-                    var outputsAttributes = pair.Value as DictionaryContextData;
-                    outputsAttributes.TryGetValue("value", out var val);
-
-                    if (val != null)
+                    var outputName = pair.Key;
+                    var outputDefinition = pair.Value as DictionaryContextData;
+                    if (outputDefinition.TryGetValue("value", out var val))
                     {
-                        var outputsValue = val as StringContextData;
-                        // Set output in the whole composite scope. 
-                        if (!String.IsNullOrEmpty(outputsValue))
-                        {
-                            ExecutionContext.SetOutput(outputsName, outputsValue, out _);
-                        }
-                        else
-                        {
-                            ExecutionContext.SetOutput(outputsName, "", out _);
-                        }
+                        var outputValue = val.AssertString("output value");
+                        ExecutionContext.SetOutput(outputName, outputValue.Value, out _);
                     }
                 }
             }
         }
 
-        private async Task RunStepsAsync(List<IStep> compositeSteps)
+        private async Task RunStepsAsync(List<IStep> nestedSteps)
         {
-            ArgUtil.NotNull(compositeSteps, nameof(compositeSteps));
+            ArgUtil.NotNull(nestedSteps, nameof(nestedSteps));
 
-            // The parent StepsRunner of the whole Composite Action Step handles the cancellation stuff already. 
-            foreach (IStep step in compositeSteps)
+            foreach (IStep step in nestedSteps)
             {
-                Trace.Info($"Processing composite step: DisplayName='{step.DisplayName}'");
-
-                step.ExecutionContext.ExpressionValues["steps"] = ExecutionContext.Global.StepsContext.GetScope(step.ExecutionContext.ScopeName);
```

Comment:
This line was previously here because of a bug. The wrong steps context was added when the step was initially created. So this line was needed to stomp over it with the correct one :)